### PR TITLE
Wasm e2e test

### PR DESCRIPTION
**Pull request by sekulicd**
This contains WASM end to end test, it is using playwright-go which is used to enable cross-browser web automation.
Basically i created minimal index.html that just loads wasm. I serve it with wasm_exec.js and ark-sdk.wasm and then i simulate Alice sends vtxos to Bob using headless browsing.
This is usefull as with this we test not only WASM but SDK as well.
GH action added.

@tiero @altafan @louisinger @bordalix please review.

### CODE REVIEW

File: pkg/client-sdk/ark-sdk.wasm.gz
Comment (tiero) [inline pkg/client-sdk/ark-sdk.wasm.gz:1]: left over?

File: pkg/client-sdk/test/wasm/web/index.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Ark SDK WASM Example</title>
+    <script src="../wasm_exec.js"></script>
```
Comment (tiero) [inline pkg/client-sdk/test/wasm/web/index.html]: is this commited?

File: .github/workflows/ark.wasm_test.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        working-directory: ./pkg/client-sdk
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v4
+        with:
+          go-version: '>=1.23.1'
+      - name: Run go work sync
+        run: go work sync
+
+      - name: Install Python
```
Comment (tiero) [inline .github/workflows/ark.wasm_test.yaml:26]: why we need python

File: .github/workflows/ark.wasm_test.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  test:
+    name: integration tests
```
Comment (tiero) [inline .github/workflows/ark.wasm_test.yaml]:
```suggestion
    name: wasm integration tests
```

File: .github/workflows/ark.wasm_test.yaml (hunk 2 above)
Comment (sekulicd) [reply .github/workflows/ark.wasm_test.yaml:26]: we serve static files with it

File: .github/workflows/ark.wasm_test.yaml (hunk 3 above)
Comment (sekulicd) [reply .github/workflows/ark.wasm_test.yaml]: Done

File: pkg/client-sdk/Makefile [hunk 4]
```diff
@@ -31,4 +31,15 @@ GO_VERSION := $(shell go version | cut -d' ' -f3)
...
+test-wasm:
+	$(MAKE) build-wasm BUILD_DIR=./test/wasm/web
+	@echo "Copying $(go env GOROOT)/misc/wasm/wasm_exec.js"
+	@cp `go env GOROOT`/misc/wasm/wasm_exec.js ./test/wasm/web
+	@echo "Starting web server..."
+	@cd ./test/wasm/web && python3 -m http.server 8000 &
+	@echo "Waiting for server to start..."
+	@sleep 3
+	@echo "Running tests..."
+	@cd ./test/wasm && go test -v
```
Comment (altafan) [inline pkg/client-sdk/Makefile:45]: once the test is done (either succeded or failed) we must clean folders and stop services

File: pkg/client-sdk/go.mod [hunk 5]
```diff
@@ -5,6 +5,8 @@ go 1.23.1
 replace github.com/btcsuite/btcd/btcec/v2 => github.com/btcsuite/btcd/btcec/v2 v2.3.3
 
 require (
+	github.com/ark-network/ark/api-spec v0.0.0-20241111144357-cd502f3dac28
+	github.com/ark-network/ark/common v0.0.0-20241111144357-cd502f3dac28
```
Comment (altafan) [inline pkg/client-sdk/go.mod]: DO NOT run go mod tidy to install deps, use go work sync instead.

ark's deps must not show up in any go mod.

File: .github/workflows/ark.unit.yaml [hunk 6]
```diff
@@ -78,7 +78,7 @@ jobs:
         with:
           version: v1.61
           working-directory: ./pkg/client-sdk
-          args: --timeout 5m
+          args: --timeout 5m --skip-files=.*_test.go
```
Comment (altafan) [inline .github/workflows/ark.unit.yaml:81]: we're skipping all test files this way while we want to skip only the wasm ones?

Comment (sekulicd) [reply .github/workflows/ark.unit.yaml:81]: do we want linting of test files?

File: pkg/client-sdk/Makefile (hunk 4 above)
Comment (sekulicd) [reply pkg/client-sdk/Makefile:45]: why? isnt GHA container destroyed after pipeline runs? i dont see we stop nigiri for eg in integration pipeline?

File: pkg/client-sdk/go.mod (hunk 5 above)
Comment (sekulicd) [reply pkg/client-sdk/go.mod]: hmm, i run go work sync in root

**Pull request closed without merging**